lib/r10k/module/svn.rb

```diff
@@ -47,7 +52,7 @@ def self.implement?(name, args)
   def initialize(name, dirname, opts, environment=nil)
     super
 
-    setopts(opts, INITIALIZE_OPTS)
+    setopts(@args, INITIALIZE_OPTS)
```

Comment:
IMO, stylistically it's better to just use the parameter name you're passed into, even it is effectively the same as an instance variable created elsewhere. I don't know if that's actually

Reply:
Yeah. The main reason this goes to `@args` is the sanitization performed on the instance variable [here](https://github.com/puppetlabs/r10k/pull/1131/files#diff-db1c390dda7d3b2f14abcee49a787a9a31ffa9a41d2a0cbea53e35fb65b86773R55), to remove a common key "type" so that individual `setopts()` calls don't have to care if it's present.

One alternative option could be to provide `setopts()` with the ability to ignore certain keys. If we did that, we'd have to ignore it everywhere, but also wouldn't have to sanitize `@args`. E.g.,

```
setopts(options, {
  # Standard option interface
  :source  => :remote,
  :version => :expected_revision,
  :type => :ignore

  # Type-specific options
  :remote => :self,
  :username => :self,
  :password => :self,
  :puppetfile_name => :self,
})
```

Think that would be better?

Reply:
Pushed a new commit with this change. Seems more straightforward, and therefore better.